src/Microsoft.FeatureManagement/ServiceCollectionExtensions.cs

```diff
@@ -27,31 +28,32 @@ public static IFeatureManagementBuilder AddFeatureManagement(this IServiceCollec
         {
             services.AddLogging();
 
+            services.AddMemoryCache();
+
             //
             // Add required services
             services.TryAddSingleton<IFeatureDefinitionProvider, ConfigurationFeatureDefinitionProvider>();
 
             services.AddSingleton(sp => new FeatureManager(
                 sp.GetRequiredService<IFeatureDefinitionProvider>(),
-                sp.GetRequiredService<IEnumerable<IFeatureFilterMetadata>>(),
-                sp.GetRequiredService<IEnumerable<ISessionManager>>(),
-                sp.GetRequiredService<ILoggerFactory>(),
-                sp.GetRequiredService<IOptions<FeatureManagementOptions>>(),
-                sp.GetRequiredService<IOptions<TargetingEvaluationOptions>>())
+                sp.GetRequiredService<IOptions<FeatureManagementOptions>>()?.Value)
             {
-                Configuration = sp.GetService<IConfiguration>(),
-                TargetingContextAccessor = sp.GetService<ITargetingContextAccessor>(),
+                FeatureFilters = sp.GetService<IEnumerable<IFeatureFilterMetadata>>(),
```

Comment:
Can you use GetRequiredService? This should ensure we always get an IEnumerable.